com.unity.multiplayer.mlapi/Runtime/Profiling/InternalMessageHandlerProfilingDecorator.cs

```diff
@@ -0,0 +1,210 @@
+﻿using System;
+using System.IO;
+using MLAPI.Messaging;
+using MLAPI.Messaging.Buffering;
+using Unity.Profiling;
+
+namespace MLAPI.Profiling
+{
+#if DEVELOPMENT_BUILD || UNITY_EDITOR
```

Comment:
nit, why bother compiling out the class since usage is already conditional? code stripping will handle removing unused classes if the user is concerned about assembly size, and compiler directives like this add compilation complexity and maintenance overhead. (not a sticking point, i know there might be some debate on this)

Reply:
That's a good point. There's a few places where we want this condition to strip out classes that we don't even want to reference, but in this case everything is in the base engine.